### PR TITLE
Solution implemented

### PR DESCRIPTION
**Pull request by NikolayAbysov**

### CODE REVIEW

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 1]
```diff
@@ -6,18 +6,50 @@
...
+        for (int i = 0; i < matrix[0].length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int [] counterDiagonal = new int[matrix[0].length];
+        int counter = matrix[0].length - 1;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: do we need this variable?

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 2]
```diff
@@ -6,18 +6,50 @@
...
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = 0;
+        if (matrix.length == 1) {
+            return matrix[0][0];
+        }
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: thats redundant check

File: src/main/java/core/basesyntax/MatrixOperation.java [hunk 3]
```diff
@@ -6,18 +6,50 @@
...
+                    min = matrix[i][j];
+                }
+            }
+        }
+        return min;
     }
 
     public int getMax(int[][] matrix) {
-        return 0;
+        int max = 0;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/MatrixOperation.java]: what if we would we have the matrix only with negative numbers?

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (NikolayAbysov) [reply src/main/java/core/basesyntax/MatrixOperation.java]: It is not redundant - it is used for single element array check:
Безымянный

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (NikolayAbysov) [reply src/main/java/core/basesyntax/MatrixOperation.java]: I have somehow to get second coordinate to move from right to left. Is it possible to do it without this variable?

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 3 above)
Comment (NikolayAbysov) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed this issue

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: I meant do you need to create a variable for it if you are using it only once and it's short enough to write it as it is.
Think about getting this value from i.

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/MatrixOperation.java]: if matrix.length would be 1 then it should go through loops above only once and still get the same matrix[0][0]. Try to do that)

File: src/main/java/core/basesyntax/MatrixOperation.java (hunk 1 above)
Comment (NikolayAbysov) [reply src/main/java/core/basesyntax/MatrixOperation.java]: Fixed this issue. Thanks for help :)

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request closed without merging**